src/REL/d/a/obj/d_a_obj_block_underground.cpp

```diff
@@ -1,6 +1,292 @@
 #include "d/a/obj/d_a_obj_block_underground.h"
 
+#include "c/c_list.h"
+#include "common.h"
+#include "d/a/d_a_player.h"
+#include "d/a/obj/d_a_obj_base.h"
+#include "d/col/bg/d_bg_s.h"
+#include "d/col/cc/d_cc_d.h"
+#include "d/col/cc/d_cc_s.h"
+#include "d/flag/sceneflag_manager.h"
+#include "d/snd/d_snd_small_effect_mgr.h"
+#include "f/f_base.h"
+#include "f/f_list_nd.h"
+#include "m/m_vec.h"
+#include "nw4r/g3d/res/g3d_resfile.h"
+#include "nw4r/g3d/res/g3d_resmdl.h"
+#include "toBeSorted/d_emitter.h"
+
 SPECIAL_ACTOR_PROFILE(OBJ_BLOCK_UNDERGROUND, dAcOBlockUnderground, fProfile::OBJ_BLOCK_UNDERGROUND, 0x13F, 0, 3);
 
-dAcOBlockUnderground::dAcOBlockUnderground() {}
-dAcOBlockUnderground::~dAcOBlockUnderground() {}
+const s32 dAcOBlockUnderground::sSubtypeIdx[12] = {0, 0, 0, 1, 1, 1, 2, 2, 2, 3, 1, 1};
```

Comment:
I wonder if this was originally an array of a struct consisting of three `u32`s, or at least an array of arrays. That would probably allow replacing the manual index calculation (`mSubtype * 3 + 1`).`